researchstack-mpower/src/main/java/org/sagebionetworks/research/mpower/researchstack/framework/step/MpPhoneInstructionStepLayout.java

```diff
@@ -43,24 +53,83 @@ public MpPhoneInstructionStepLayout(Context context, AttributeSet attrs, int def
     public void goForwardClicked(View v) {
         MpDataProvider provider = MpDataProvider.getInstance();
         String phoneNumber = phoneEntryField.getText().toString();
-        Phone phone = new Phone().number(phoneNumber).regionCode("US");
 
-        Subscription response = provider.signUp(phone)
+        findViewById(R.id.button_go_forward)
+                .setEnabled(false);
+
+        Observable<DataResponse> phoneAuthObservable;
+
+        // check whether we should create new accounts
+        if (getResources().getBoolean(R.bool.should_sign_up_phone)) {
+            LOGGER.debug("Doing phone sign up and sign in");
+            Phone phone = new Phone().regionCode("US").number(phoneNumber);
```

Comment:
Hard-coded string, should this be in the "Phone" class?

Reply:
Phone class is autogenerated from Bridge API. I could make a constant